tests/utils/streams/test_batching.py

```diff
@@ -1,99 +1,102 @@
 import time
 from datetime import datetime
+from typing import Any, MutableSequence, Sequence, Optional
 from unittest.mock import patch
 
+from confluent_kafka import Message
+
 from snuba.utils.metrics.backends.dummy import DummyMetricsBackend
 from snuba.utils.streams.batching import AbstractBatchWorker, BatchingKafkaConsumer
 from tests.backends.confluent_kafka import FakeConfluentKafkaConsumer, FakeConfluentKafkaMessage, FakeConfluentKafkaProducer
 
 
-class FakeBatchingKafkaConsumer(BatchingKafkaConsumer):
-    def create_consumer(self, *args, **kwargs):
-        return FakeConfluentKafkaConsumer()
-
-
 class FakeWorker(AbstractBatchWorker):
-    def __init__(self, *args, **kwargs):
-        super(FakeWorker, self).__init__(*args, **kwargs)
-        self.processed = []
-        self.flushed = []
+    def __init__(self) -> None:
+        self.processed: MutableSequence[Optional[Any]] = []
+        self.flushed: MutableSequence[Sequence[Any]] = []
 
-    def process_message(self, message):
+    def process_message(self, message: Message) -> Optional[Any]:
         self.processed.append(message.value())
         return message.value()
 
-    def flush_batch(self, batch):
+    def flush_batch(self, batch: Sequence[Any]) -> None:
         self.flushed.append(batch)
 
 
 class TestConsumer(object):
-    def test_batch_size(self):
-        consumer = FakeBatchingKafkaConsumer(
+    def test_batch_size(self) -> None:
+        consumer = FakeConfluentKafkaConsumer()
+        worker = FakeWorker()
+        producer = FakeConfluentKafkaProducer()
```

Comment:
These have to be instantiated outside of the constructor so that there is a local variable reference.

Otherwise, lines like `assert batching_consumer.worker.processed` won't (and shouldn't) type check correctly since the abstract worker does not expose a `processed` property or attribute.